iib/workers/tasks/build_containerized_merge.py

```diff
@@ -0,0 +1,368 @@
+        set_request_state(request_id, 'in_progress', 'Processing the deprecation list')
+        intermediate_bundles = missing_bundle_paths + source_index_bundles_pull_spec
+        deprecation_bundles = get_bundles_from_deprecation_list(
+            intermediate_bundles, deprecation_list
+        )
+        deprecation_bundles = deprecation_bundles + [
+            bundle['bundlePath'] for bundle in invalid_bundles
+        ]
+
+        # process the deprecation list into the intermediary index.db file
+        if deprecation_bundles:
+            # We need to get the latest pullpecs from bundles in order to avoid failures
+            # on "opm deprecatetruncate" due to versions already removed before.
+            # Once we give the latest versions all lower ones get automatically deprecated by OPM.
+            all_bundles = source_index_bundles + target_index_bundles
+            deprecation_bundles = get_bundles_latest_version(deprecation_bundles, all_bundles)
+
+            deprecate_bundles_db(
+                base_dir=temp_dir, index_db_file=source_index_db_path, bundles=deprecation_bundles
+            )
+
+        # Retrieve the operators from the intermediary index.db file
+        # This will be required for pushing the updated index.db file to the IIB registry
+        bundles_in_db = get_list_bundles(source_index_db_path, temp_dir)
+        operators_in_db = [bundle['packageName'] for bundle in bundles_in_db]
+
+        # Migrate the intermediary index.db file to FBC and generate the Dockerfile
+        set_request_state(
+            request_id,
+            'in_progress',
+            'Migrating the intermediary index.db file to FBC and generating the Dockerfile',
+        )
+        fbc_dir, _ = opm_migrate(source_index_db_path, temp_dir)
+
+        # rename `catalog` directory because we need to use this name for
+        # final destination of catalog (defined in Dockerfile)
+        catalog_from_db = os.path.join(temp_dir, 'from_db')
+        os.rename(fbc_dir, catalog_from_db)
+
+        # Merge migrated FBC with existing FBC in Git repo
+        # overwrite data in `catalog_from_index` by data from `catalog_from_db`
+        # this adds changes on not opted in operators to final FBC
+        log.info('Merging migrated catalog with Git catalog')
+        merge_catalogs_dirs(catalog_from_db, localized_git_catalog_path)
+
+        # We need to regenerate file-based catalog because we merged changes
+        fbc_dir_path = os.path.join(temp_dir, 'catalog')
+        if os.path.exists(fbc_dir_path):
+            shutil.rmtree(fbc_dir_path)
+        shutil.move(catalog_from_db, fbc_dir_path)
+
+        # Validate the FBC config
+        set_request_state(request_id, 'in_progress', 'Validating the FBC config')
+        opm_validate(fbc_dir_path)
+
+        # Write build metadata to a file to be added with the commit
+        set_request_state(request_id, 'in_progress', 'Writing build metadata')
+        write_build_metadata(
```

Comment:
Please also include `arches` when calling this function. The change to the function signature is here: 5d59cd78bb0796b5c3089e0b0ba5d6b51e426016